### PR TITLE
improve(SpokePoolClient): Skip unnecessary LP fee computation

### PR DESCRIPTION
**Pull request by pxrl**
The SpokePoolClient currently piggybacks on the HubPoolClient's batch LP fee computation in order to optimise resolving quoteTimestamp to a mainnet block number. This was historically necessary prior to the v3 transition, but it's no longer needed now that the LP fee is computed by upper-layer code. Dropping the LP fee computation will reduce RPC queries on update.

### CODE REVIEW

File: src/clients/HubPoolClient.ts [hunk 1]
```diff
@@ -291,11 +291,33 @@ export class HubPoolClient extends BaseAbstractClient {
...
+   * @dev Querying block numbers sequentially also improves BlockFinder efficiency.
+   * @param timestamps Array of timestamps to be resolved to a block number on the HubPool chain.
+   * @returns A mapping of quoteTimestamp -> HubPool block number.
+   */
+  async getBlockNumbers(timestamps: number[]): Promise<{ [quoteTimestamp: number]: number }> {
+    const sortedTimestamps = dedupArray(timestamps).sort((x, y) => x - y);
+    const blockNumbers: { [quoteTimestamp: number]: number } = {};
+    for (const timestamp of sortedTimestamps) {
+      blockNumbers[timestamp] = await this.getBlockNumber(timestamp);
+    }
```
Comment (dohaki) [inline src/clients/HubPoolClient.ts:316]: NIT: Can this be wrapped in a `Promise.all`?

**Review by dohaki — APPROVED** [review]
Looking good 👍 Left one nit / q

**Review by bmzig — APPROVED** [review]
I think this makes sense.

File: src/clients/HubPoolClient.ts (hunk 1 above)
Comment (pxrl) [reply src/clients/HubPoolClient.ts:316]: You'd think so, but it's deliberately done this way to allow the BlockFinder to handle these one by one. If we hit the BlockFinder in parallel then it doesn't get any of the benefits of previous cached blocks, so we risk kicking off lots of parallel RPC queries. If we hit it sequentially with the oldest timestamp first then it should be able to infer the correct block number on subsequent requests very accurately based on the fairly stable block times on mainnet.

**Pull request merged**